apps/website-25/src/components/courses/exercises/Exercise.tsx

```diff
@@ -12,23 +15,64 @@ type ExerciseProps = {
 const Exercise: React.FC<ExerciseProps> = ({
   exerciseId,
 }) => {
-  const [{ data }] = useAxios<GetExerciseResponse>({
+  const exerciseClassNames = 'exercise not-prose my-6';
+
+  const auth = useAuthStore((s) => s.auth);
+
+  const [{ data: exerciseData }] = useAxios<GetExercise>({
     method: 'get',
     url: `/api/courses/exercises/${exerciseId}`,
   });
 
-  if (!data || !data.exercise) {
+  const [{ data: responseData }, refetch] = useAxios<GetExerciseResponseResponse>({
```

Comment:
future: I think it would be good if we don't try to get a exercise response at all if not logged in. I think you can achieve this with { manual: auth == undefined }, because then if auth is not defined it won't automatically make the request when loading the component (I might be wrong).

Appreciate you can't do this with an if because of the rules of react hooks.